### PR TITLE
Home work 6 (Oleg Zadniprovskyi)

### PR DESCRIPTION
**Pull request by gelono**

### CODE REVIEW

File: books/tasks.py [hunk 1]
```diff
@@ -42,3 +43,24 @@ def send_customer_email(order_id):
...
+@shared_task(autoretry_for=(BaseException,), retry_backoff=2)
+def new_order_created(order_id):
+    instance = OrderLineItem.objects.get(id=order_id)
+    book_name = instance.book.name
+    quantity = instance.quantity
+    b: Book = Book.objects.get(name=book_name)
+    authors = b.authors.all()
+    for a in authors:
+        text = f'"Hi, {a.first_name}! You have a your book sold ({quantity} inst.). Sincerely, your Oleg Zadniprovskyi"'
+        send_telegram_message(text, a.telegram_account_id)
```
Comment (p1v2) [inline books/tasks.py]: what will happen if telegram_account_id is None? I believe your function will fail, but it shouldn't

File: books/tasks.py [hunk 2]
```diff
@@ -42,3 +43,24 @@ def send_customer_email(order_id):
...
+
+
+@shared_task
+def statistic_sending():
+    authors = Author.objects.annotate(total_count_sold=Sum('book__count_sold'))
+    for author in authors:
+        if author.telegram_account_id:
+            text = f'Hi, {author.first_name}! There is already {author.total_count_sold} of your books being sold"'
+            send_telegram_message(text, author.telegram_account_id)
```
Comment (p1v2) [inline books/tasks.py:68]: same here

**Review by p1v2 — CHANGES_REQUESTED** [review]
Pls check my comments

File: books/tasks.py (hunk 1 above)
Comment (gelono) [reply books/tasks.py]: Yes, there is a lack of validation

File: books/tasks.py (hunk 2 above)
Comment (gelono) [reply books/tasks.py:68]: Row #64. Is this validation not enough? If it's True, the message will be sent. If it's False (None), the message won't be sent.

Comment (p1v2) [reply books/tasks.py:68]: aha, I see here. Ok, here it fine. then pls check my previous comment

**Review by p1v2 — APPROVED** [review]

**Pull request closed without merging**